### ISSUE TITLE
Eggs not incubated, incubateEggs fails

### ISSUE THREAD

**Issue by kenkclam** [PokemonGoF/PokemonGo-Bot#4461]
### Expected Behaviour
Bot incubate eggs

### Actual Behaviour
There are no messages on the log mentioning egg incubation or egg hatching whatsoever
On the web UI, it said NaN/NaNkm under the incubator

### Your FULL config.json (remove your username, password, gmapkey and any other private info)
{
    "auth_service": "",
    "username": "",
    "password": "",
    "location": "",
    "gmapkey": "",
    "libencrypt_location": "",
    "websocket_server": true,
    "heartbeat_threshold": 11,
    "tasks": [
      {
        "type": "MoveToMapPokemon",
        "config": {
          "address": "http://localhost:5000",
          "max_distance": 1000,
          "min_time": 10,
          "min_ball": 20,
          "prioritize_vips": true,
          "snipe": false,
          "snipe_high_prio_threshold": 4000,
          "update_map": false,
          "mode": "priority",
          "catch": {
            "Poliwag": 10,
            "Nidoran F": 10,
            "//Nidoran M": 10,
            "Clefairy": 15,
            "Bellsprout": 10,
            "Slowpoke": 10,
            "Eevee": 20,
            "Oddish": 20,
            "Geodude": 50,
            "Abra": 50,
            "Diglett": 100,
            "Vulpix": 100,
            "Kabuto": 100,
            "Muk": 300,
            "Poliwrath": 200,
            "Golem": 200,
            "Weezing": 300,
            "Electrode": 300,
            "Primeape": 300,
            "Mr-mimi": 1000,
            "Chansey": 1000,
            "Alakazam": 600,
            "Omastar": 500,
            "Kabutops": 500,
            "Raichu": 600,
            "Tauros": 500,
            "Dugtrio": 500,
            "Bulbasaur": 400,
            "Charmander": 500,
            "Charmeleon": 600,
            "Squirtle": 300,
            "Pikachu": 600,
            "Growlithe": 400,
            "Machop": 300,
            "Grimer": 200,
            "Exeggcute": 200,
            "Omanyte": 100,
            "Koffing": 200,
            "Dratini": 600,
            "Lapras": 1000,
            "Electabuzz": 1000,
            "Magmar": 800,
            "Ditto": 900,
            "Arcanine": 700,
            "Machamp": 700,
            "Blastoise": 800,
            "Venusaur": 800,
            "Charizard": 900,
            "Slowbro": 300,
            "Nidoqueen": 150,
            "Nidoking": 150,
            "Clefable": 100,
            "Victreebel": 200,
            "Jolteon": 200,
            "Magneton": 100,
            "Gengar": 100,
            "Machoke": 80,
            "Vileplume": 400,
            "Exeggutor": 600,
            "Aerodactyl": 600,
            "Snorlax": 1200,
            "Articuno": 1200,
            "Zapdos": 1200,
            "Moltres": 1200,
            "Dragonair": 1000,
            "Dragonite": 1000,
            "Mewtwo": 1300,
            "Mew": 1300
          }
        }
      },
      {
        "type": "SleepSchedule",
        "config": {
          "enabled": true,
          "time": "00:00",
          "duration": "7:00",
          "time_random_offset": "00:22",
          "duration_random_offset": "00:43"
        }
      },
      {
        "type": "HandleSoftBan"
      },
      {
        "type": "CollectLevelUpReward"
      },
      {
        "type": "IncubateEggs",
        "config": {
          "longer_eggs_first": true
        }
      },
      {
        "type": "TransferPokemon"
      },
      {
        "type": "EvolvePokemon",
        "config": {
            "evolve_all": "Pidgey,Rattata,Weedle,Caterpie",
            "first_evolve_by": "cp",
            "evolve_above_cp": 200,
            "evolve_above_iv": 0.2,
            "logic": "and",
            "evolve_speed": 20,
            "use_lucky_egg": false
        }
      },
      {
        "type": "RecycleItems",
        "config": {
          "item_filter": {
            "Pokeball":       { "keep" : 10 },
            "Potion":         { "keep" : 0 },
            "Super Potion":   { "keep" : 0 },
            "Hyper Potion":   { "keep" : 80 },
            "Max Potion":     { "keep" : 100 },
            "Revive":         { "keep" : 0 },
            "Max Revive":     { "keep" : 40},
	    "Greatball":      { "keep" : 30 },
            "Ultraball":      { "keep" : 50 },
            "Razz Berry":     { "keep" : 20 }
          }
        }
      },
	    {
	      "type": "CatchPokemon",
	      "config": {
	        "catch_visible_pokemon": true,
	        "catch_lured_pokemon": true,
	        "min_ultraball_to_keep": 2,
	        "catch_throw_parameters": {
	          "excellent_rate": 0.1,
	          "great_rate": 0.2,
	          "nice_rate": 0.6,
	          "normal_rate": 0.1,
	          "spin_success_rate" : 0.6
	        },
				  "catch_simulation": {
				    "flee_count": 2,
				    "flee_duration": 2,
				    "catch_wait_min": 1,
				    "catch_wait_max": 3,
				    "berry_wait_min": 2,
				    "berry_wait_max": 3,
				    "changeball_wait_min": 2,
				    "changeball_wait_max": 3
				  }
	      }
	    },
      {
        "type": "SpinFort",
        "config": {
          "spin_wait_min": 2,
          "spin_wait_max": 8
           }
      },
      {
        "type": "MoveToFort",
        "config": {
            "lure_attraction": true,
            "lure_max_distance": 500
        }
      },
      {
        "type": "FollowSpiral",
        "config": {
          "diameter": 1,
          "step_size": 5
        }
      }
    ],
    "map_object_cache_time": 5,
    "forts": {
      "avoid_circles": true,
      "max_circle_size": 10
    },
    "walk_max": 6,
    "walk_min": 3,
    "action_wait_min": 1,
    "action_wait_max": 4,
    "debug": false,
    "test": false,
    "health_record": true,
    "location_cache": false,
    "distance_unit": "km",
    "reconnecting_timeout": 30,
    "evolve_captured": "NONE",
    "catch_randomize_reticle_factor": 1.0,
    "catch_randomize_spin_factor": 1.0,
    "catch": {
      "any": {"catch_above_cp": 0, "catch_above_iv": 0, "logic": "or"},
      "Rattata": {"never_catch": true},
      "Weedle": {"never_catch": true},
      "//Pidgey": {"never_catch": true},
      "//Staryu": {"never_catch": true},
      "//Starmie": {"never_catch": true},
      "//Goldeen": {"never_catch": true},
      "//Drowzee": {"never_catch": true},
      "Spearow": {"never_catch": true},
      "Zubat": {"never_catch": true},
      "Krabby": {"never_catch": true},
      "//Pidgeotto": {"never_catch": true}
    },
    "release": {
      "any": {"release_below_cp": 550, "release_below_iv": 0.80, "logic": "and"},
      "Bulbasaur": {"keep_best_cp": 1, "keep_best_iv": 2},
      "Squirtle": {"keep_best_cp": 1, "keep_best_iv": 2},
      "Nidoran M": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Clefairy": {"keep_best_cp": 1, "keep_best_iv": 2},
      "Rattata": {"keep_best_cp": 1},
      "Pidgey": {"keep_best_cp": 2, "keep_best_iv": 1},
      "Starmie": {"release_below_cp": 900, "release_below_iv": 0.80, "logic": "and"},
      "Raticate": {"release_below_cp": 900, "release_below_iv": 0.85, "logic": "and"},
      "Pidgeotto": {"release_below_cp": 900, "release_below_iv": 0.85, "logic": "and"},
      "Metapod": {"release_below_cp": 350, "release_below_iv": 0.85, "logic": "and"},
      "Kakuna": {"release_below_cp": 400, "release_below_iv": 0.85, "logic": "and"},
      "Zubat": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Caterpie": {"keep_best_cp": 1},
      "Weedle": {"keep_best_cp": 1},
      "Spearow": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Sandshrew": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Nidoran F": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Jigglypuff": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Oddish": {"keep_best_cp": 1, "keep_best_iv": 2},
      "Paras": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Venonat": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Vulpix": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Mankey": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Diglett": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Abra": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Meowth": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Psyduck": {"keep_best_cp": 2, "keep_best_iv": 2},
      "Poliwag": {"keep_best_cp": 1, "keep_best_iv": 3},
      "Poliwhirl": {"keep_best_cp": 1, "keep_best_iv": 2},
      "Geodude": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Ponyta": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Tentacool": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Bellsprout": {"keep_best_cp": 1, "keep_best_iv": 2},
      "Slowpoke": {"keep_best_cp": 1, "keep_best_iv": 3},
      "Seel": {"release_below_cp": 650, "release_below_iv": 0.85, "logic": "and"},
      "Exeggcute": {"keep_best_cp": 1, "keep_best_iv": 2},
      "Magnemite": {"keep_best_cp": 1, "keep_best_iv": 2},
      "Shellder": {"keep_best_cp": 1, "keep_best_iv": 2},
      "Doduo": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Dodrio": {"release_below_cp": 900, "release_below_iv": 0.7, "logic": "and"},
      "Tangela": {"release_below_cp": 800, "release_below_iv": 0.8, "logic": "and"},
      "Scyther": {"release_below_cp": 800, "release_below_iv": 0.8, "logic": "and"},
      "Pinsir": {"release_below_cp": 1600, "release_below_iv": 0.8, "logic": "and"},
      "Gastly": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Drowzee": {"keep_best_cp": 1, "keep_best_iv": 3},
      "Rhyhorn": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Krabby": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Cubone": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Koffing": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Horsea": {"keep_best_cp": 1, "keep_best_iv": 2},
      "Goldeen": {"keep_best_cp": 1, "keep_best_iv": 2},
      "Staryu": {"keep_best_cp": 1, "keep_best_iv": 2},
      "Magikarp": {"keep_best_cp": 1, "keep_best_iv": 2},
      "Dratini": {"keep_best_cp": 1, "keep_best_iv": 2},
      "Eevee": {"keep_best_cp": 2, "keep_best_iv": 3},
      "Ekans": {"keep_best_cp": 1, "keep_best_iv": 1},
      "Growlithe": {"keep_best_cp": 2, "keep_best_iv": 2},
      "Jynx": {"keep_best_cp": 2},
      "Dodrio": {"keep_best_cp": 2}
    },
    "vips" : {
         "Any pokemon put here directly force to use Berry & Best Ball to capture, to secure the capture rate!": {},
        "any": {"catch_above_cp": 1200, "catch_above_iv": 0.9, "logic": "or" },
        "Lapras": {},
        "Moltres": {},
        "Zapdos": {},
        "Articuno": {},

        "// S-Tier pokemons (if pokemon can be evolved into tier, list the representative)": {},
        "Mewtwo": {},
        "Dragonite": {},
        "Snorlax": {},
        "// Mew evolves to Mewtwo": {},
        "Mew": {},
        "Arcanine": {},
        "Vaporeon": {},
        "Gyarados": {},
        "Exeggutor": {},
        "Muk": {},
        "Weezing": {},
        "Flareon": {}

    }
}

### Output when issue occurred
no mentioning of egg incubation

### Steps to Reproduce
It was working before, I guess I must have done something silly while modifying my config
But I cannot figure it out, please help

### Other Information
OS: 
 
Branch: 
 
Git Commit: 
 
Python Version: 
 
Any other relevant files/configs (eg: path files)

**Comment by mjmadsen** [comment]
This might be better suited for https://github.com/PokemonGoF/PokemonGo-Web

**Comment by kenkclam** [comment]
why is it more suitable for web UI? 

I open my account on my cell phone, eggs are not put in incubator. 
There is no sign of egg incubation on the bot log, web UI nor on my iPhone

**Comment by mjmadsen** [comment]
What commit are you using?

I said it was related to the UI because you referenced the UI.

**Comment by Gurzeh** [comment]
same: not incubating eggs, but my commit is old:
e63074be072b6956ba0c37156bcd83792469bb16

**Comment by kenkclam** [comment]
I did 
git pull
pip install -r requirements.txt --upgrade

Now it seems to start incubating egg again
I did some major modifications to the config file and then this problem started
I thought I must have done something silly there.
Turned out the config is completely fine, I just need to update the bot (but it was working fine before. .. well, whatever). 


Thanks for the troubleshooting. I am closing the issue.
